### DISCUSSION TITLE
Nested vec

### DISCUSSION THREAD

**Discussion by sondrelg** [sharksforarms/deku#336] [Q&A] [answered]
Hi!

Was just wondering if anyone knows a way to handle `Vec<Vec<u8>>` with deku? I see from the documentation that a `Vec<u8>` is representable with a count variable (and I've gotten this to work), but I'm struggling to make progress on the nested vec case.

To avoid the x y problem; what I'm really trying to do, is represent `Vec<String>`. If there's another way to do that, that would also be helpful :+1: 

Thanks!

**Comment by sondrelg** [comment]
I guess I could achieve what I'm trying to do by joining the `Vec<String>` into a single string and representing it as `Vec<u8>`

**Comment by sondrelg** [comment, marked as answer]
I found this to be a little simpler:

```rust
#[derive(Debug, PartialEq, DekuRead, DekuWrite)]
pub struct DekuString {
    #[deku(bytes = "1")]
    content_len: u8,
    #[deku(count = "content_len")]
    content: Vec<u8>,
}

impl Into<String> for DekuString {
    fn into(self) -> String {
        String::from_utf8_lossy(&self.content).to_string()
    }
}

impl From<String> for DekuString {
    fn from(value: String) -> Self {
        // TODO: Handle u8 conversion
        DekuString { content_len: value.len() as u8, content: Vec::from(value) }
    }
}
```

and then doing:

```rust
#[derive(Debug, PartialEq, DekuRead, DekuWrite)]
pub struct Foo {
    #[deku(bytes = "1")]
    foo_len: u8,
    #[deku(count = "foo_len")]
    foo: Vec<DekuString>,
}
```

Please let me know if there are better ways :bow:

**Reply by wcampbell0x2a** [reply]
Only comment I have is that you don't need `    #[deku(bytes = "1")]` for just a u8.
